backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/controllers/BookController.java

```diff
@@ -0,0 +1,77 @@
+package com.nokia.library.nokiainnovativeproject.controllers;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.validation.Valid;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.nokia.library.nokiainnovativeproject.entities.Book;
+import com.nokia.library.nokiainnovativeproject.exceptions.ResourceNotFoundException;
+import com.nokia.library.nokiainnovativeproject.repositories.BookRepository;
+
+@RestController
+@RequestMapping("/library")
+public class BookController {
+
+	@Autowired
+	private BookRepository bookRepository;
```

Comment:
Please read about MVC and three-tier architecture. Then please refactor code to achieve this architecture.